### DISCUSSION TITLE
What should be the display name semantics for person?

### DISCUSSION THREAD

**Discussion by samisa-abeysinghe** [Avinya-Foundation/public-issues#6] [Ideas]
$subject given that we capture preferred name and full name for person

**Comment by rukmal** [comment]
The global data service will capture `preferred_name` and `full_name`. We will get the `last_name` by taking the last word from `full_name`, and what will be displayed should be `preferred_name`<space>`last_name`.

**Reply by samisa-abeysinghe** [reply]
`name_with_inititials` is also computed?

And GraphQL API will have `display_name` and `name_with_inititials` pre-computed?

**Reply by rukmal** [reply]
Yes, I can add those fields to the response.

**Comment by samisa-abeysinghe** [comment]
Excellent!

**Comment by sanjiva** [comment]
My thoughts on this:

Note that I'm using "given name" as the name that people use.

Names are very culture dependent. E.g.:
- Sinhalaese: Family (aka gay) names, first name, last name from family
- Tamil: Father's name, first name
- Muslim: (not certain) Family names, first name, last name from family

So, we should have IMO:
- full name
- given name (ask the person for it because we can't assume it - e.g. my younger sis uses J T Sanathani V Weerawarana .. and I have J T V Sanjiva Weerawarana (the V is the same by chance - we have 2 given names, 2 family names and the family last name)
And not do any "last name" concept for LK names. For practical purposes (like sorting) it is useful to have a last name but we should not use it to address the person.

Additional reference: https://www.worldgenweb.org/lkawgw/nametrad.htm

**Comment by samisa-abeysinghe** [comment]
One other way to easily solve this would be to simply ask for the first name and the last name of the person. Because, they will have to figure this out when they travel, apply for foreign education or studies, and the like. Then they will have to do the mapping and fill out the forms. 

I still need an answer to my original question on the display name. 
Note that the display name is for the digital systems. For transcripts, certificates, contracts, and other official documents of the organization the full name will have to be used.

**Reply by sanjiva** [reply]
IMO display name should be just be the given name. 

The problem is that the first/last split is a western concept only and doesn't fit the naming systems of several of our cultures. That way we just say: 
- we need your full legal name (of course)
- we need the name to call you by (given name or nick name or whatever you want)
The second has no legal or formal use and is only used to use as the display name and for us to talk to you with.

**Reply by samisa-abeysinghe** [reply]
I was thinking more of the students being prepared for travel and foreign jobs in the future. As you and I figured when we traveled initially, they will have to figure this out at a time when they fill out a visa application or an immigration form. 

Anyway, for the display name, we can live with `given_name`

**Comment by samisa-abeysinghe** [comment]
IDP SCIM uses givenName and familyName
Hence, for both Google and Asgardeo, we need an algorithm to map to givenName and familyName from full name and preferred name

**Comment by rukmal** [comment]
Discussion outcome with @samisa-abeysinghe @sanjiva and I:

- Text box to take preferred_name and full_name
- Drop down to select family_name from a selection of full_name split on spaces
